cmd/api/src/queries/graph.go

```diff
@@ -382,20 +386,23 @@ func (s *GraphQuery) RawCypherSearch(ctx context.Context, rawCypher string) (mod
 
 			return nil
 		}, func(config *graph.TransactionConfig) {
-			if !s.DisableCypherQC {
-				// Start at 30,000 milliseconds
-				availableRuntime := float64(30_000)
+			// Rely on the context timeout to set our query upper-bound
+			availableRuntime := bhCtxInst.Timeout.Value
 
-				// The weight of the query is divided by 5 to get a runtime reduction factor. This means that query weights
-				// of 5 or less will get the full 30 seconds of runtime.
-				reductionFactor := preparedQuery.complexity.Weight / 5
+			log.Infof("Available timeout for query is set to: %.2f seconds", availableRuntime.Seconds())
 
-				// Reduce the available runtime
-				availableRuntime /= reductionFactor
+			if !s.DisableCypherQC && !bhCtxInst.Timeout.UserSet {
```

Comment:
Do we have any automated tests that verify this timeout value is being exercise correctly? I did a quick skim and did not see one, but I could have missed it. Whether we do or don't, do you think it would be valuable to add a test that also considers the `Prefer` header use case?

Reply:
Let me jam on this and update the changeset accordingly.

Reply:
I've added three test cases that validate the inner QC timeout and user-preference overrides.